### PR TITLE
Add .gitignore, fix phpDoc and a mistake on a static call.

### PR DESCRIPTION
**Pull request by Metalaka**

### CODE REVIEW

File: Exception.php [hunk 1]
```diff
@@ -55,10 +55,10 @@ class Exception extends Idle implements Event\Source
...
      * chaining, we can add a previous exception.
      *
-     * @param   string          $message      Formatted message.
-     * @param   int             $code         Code (the ID).
-     * @param   array           $arguments    Arguments to format message.
-     * @param   \BaseException  $previous     Previous exception in chaining.
+     * @param   string      $message      Formatted message.
+     * @param   int         $code         Code (the ID).
+     * @param   array       $arguments    Arguments to format message.
+     * @param   \Exception  $previous     Previous exception in chaining.
```
Comment (Hywan) [inline Exception.php]: Nop, it's `BaseException`.

File: Idle.php [hunk 2]
```diff
@@ -262,9 +262,9 @@ public function raise($previous = false)
     /**
      * Catch uncaught exception (only \Hoa\Exception\Idle and children).
      *
-     * @param   \BaseException  $exception    The exception.
+     * @param   \Exception  $exception    The exception.
```
Comment (Hywan) [inline Idle.php]: Same here.

File: Idle.php [hunk 3]
```diff
@@ -262,9 +262,9 @@ public function raise($previous = false)
...
      * @return  void
-     * @throws  \BaseException
+     * @throws  \Exception
```
Comment (Hywan) [inline Idle.php]: Same here.

File: Idle.php [hunk 4]
```diff
@@ -307,7 +307,7 @@ public static function enableUncaughtHandler($enable = true)
         }
 
         return set_exception_handler(function ($exception) {
-            return Exception\Idle::uncaught($exception);
+            return self::uncaught($exception);
```
Comment (Hywan) [inline Idle.php:310]: Is `self` defined here in PHP5.4+? If yes, :+1:.

File: Error.php [hunk 5]
```diff
@@ -84,7 +84,7 @@ public static function enableErrorHandler($enable = true)
         }
 
         return set_error_handler(
-            function ($no, $str, $file = null, $line = null, $ctx = null) {
```
Comment (Hywan) [inline Error.php:87]: :-1:, `$ctx` exists and is “required”, even if we don't use it.

File: Exception.php (hunk 1 above)
Comment (Metalaka) [inline Exception.php]: `BaseException` only exists on non-stable versions of php7 ([rfc](https://wiki.php.net/rfc/throwable-interface), [manual](http://php.net/manual/en/language.errors.php7.php))

Comment (Hywan) [inline Exception.php]: But why should we not write an up-to-date (regarding PHP VM) API documentation?

File: Idle.php (hunk 4 above)
Comment (Metalaka) [inline Idle.php:310]: Yep works on 5.4+ https://3v4l.org/oG9QP

Comment (Hywan) [inline Idle.php:310]: :+1:

File: Exception.php (hunk 1 above)
Comment (Metalaka) [inline Exception.php]: So go for `Throwable` ?

Comment (Hywan) [inline Exception.php]: Go :-).

**Comment by Metalaka** [comment]
Rebased.

**Comment by Hywan** [comment]
Thanks!

**Pull request merged**